python/ray/worker.py

```diff
@@ -1523,6 +1523,7 @@ def process_task(task): # wrapping these lines in a function should cause the lo
                                         "function_name": function_name})
 
   check_main_thread()
+  worker.photon_client.register_pid()
```

Comment:
Also, the purpose of this should be documented (e.g., "We pass the process ID to the local scheduler so that the local scheduler can terminate this worker if it wants to.")